File: 1889-check-if-number-is-a-sum-of-powers-of-three/1889-check-if-number-is-a-sum-of-powers-of-three.py
class Solution:
    def checkPowersOfThree(self, n: int) -> bool:
        
        lis = list()
        lis.append(1)
        value = 1
        for i in range(1,17):
            value *=3
            lis.append(value)
        # print(lis)
        for i in range(len(lis)-1,-1,-1):
            if n >= lis[i]:
                n-=lis[i]
                if n==0:
                    return True
        return False

File: 1889-check-if-number-is-a-sum-of-powers-of-three/1889-check-if-number-is-a-sum-of-powers-of-three.java
class Solution {
    public boolean checkPowersOfThree(int n) {
        int start = 1;
        
        List <Long> power_of_3 = new ArrayList<Long>();
        
        power_of_3.add(1L);
        long value = 1;
        for(int i = 1;i<17;i++){
            value *=3;
            power_of_3.add(value);
            
        }
        // System.out.println(power_of_3);
        for(int i = power_of_3.size()-1;i>=0;i--){
            if(n >= power_of_3.get(i)){
                n-= power_of_3.get(i);
                // System.out.println(n+" "+ power_of_3.get(i));
                if(n==0){
                    return true;
                }
            }
        }
        return false;
    }
}